packages/utils/src/ratelimit.ts

```diff
@@ -78,14 +78,17 @@ export function updateRateLimits(
      *         Only present if rate limit applies to the metric_bucket data category.
      */
     for (const limit of rateLimitHeader.trim().split(',')) {
-      const [retryAfter, categories] = limit.split(':', 2);
+      const [retryAfter, categories, , , namespaces] = limit.split(':', 5);
       const headerDelay = parseInt(retryAfter, 10);
       const delay = (!isNaN(headerDelay) ? headerDelay : 60) * 1000; // 60sec default
       if (!categories) {
         updatedRateLimits.all = now + delay;
       } else {
         for (const category of categories.split(';')) {
-          updatedRateLimits[category] = now + delay;
+          // namespaces will be present when category === 'metric_bucket'
+          if (category !== 'metric_bucket' || !namespaces || namespaces.split(';').includes('custom')) {
+            updatedRateLimits[category] = now + delay;
+          }
```

Comment:
```suggestion
          if (category === 'metric_bucket') {
            // namespaces will be present when category === 'metric_bucket'
            if (!namespaces || namespaces.split(';').includes('custom')) {
              updatedRateLimits[category] = now + delay;
            }
          } else {
            updatedRateLimits[category] = now + delay;
          }
```

This is giga nitting but I would do it like this. Regardless of bundlesize (hoping terser is smart enough). Makes it easier to understand this admittedly convoluted logic.